addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1432,6 +1432,13 @@ export class OdooEditor extends EventTarget {
         if (!range) return;
         let start = range.startContainer;
         let end = range.endContainer;
+        const table = closestElement(sel.anchorNode, 'table') || closestElement(sel.focusNode, 'table');
+        if (table && table.nextSibling &&
+            range.intersectsNode(table.nextElementSibling) &&
+            range.intersectsNode(firstLeaf(table)))
+        {
+            range.setStartBefore(table);
+        }
```

Comment:
Isn't there the same problem but in reverse ? So something like this:
```
som[e text<table>...<td>This is the last TD.]</td></table>
```
Is there the same bug in that case ? If yes, is it also fixed by your changes here ? Because they seem to be very targeted to the very specific problem we are having, as it looks for the next sibling etc.

Reply:
@dmo-odoo This was never a problem the table would be deleted in this case.

Reply:
That is interesting, in fact. Do you know why it works in one case but not the other ? I would have expected the code to work the same way in both cases, without needing to explicitely reference any of the two situations.

Reply:
The end blocks are removed [here](https://github.com/odoo/odoo/blob/15.0/addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js#L1473), but for `start` passing `false` argument in `isRemovableInvisible(start)` would remove all the block elements at start. so a specific condition for table?
```
+        const table = closestElement(start, 'table');
+        const noBlocks = (table && firstLeaf(table) === start && !table.contains(end)) ? false : true;
         while (
             start &&
-            isRemovableInvisible(start) &&
+            isRemovableInvisible(start, noBlocks) &&
             !(endIsStart && start.contains(range.startContainer))
         ) {
```

Reply:
@deso-odoo I checked in 16.0 and the bug is not reproducible there so I think a special case for this in 15.0 is probably the best way to go, yes. Thanks !

Reply:
@dmo-odoo OK :+1: I'll update the code and move the task to RD Testing IN, so we can test it once again.

Reply:
Nice, thanks ! 👍